pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -241,6 +241,11 @@ func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machi
 
 	ec2svc := ec2.NewService(scope.Scope)
 
+	if scope.MachineStatus.InstanceID == nil {
```

Comment:
I think it'll be better to perform this check in the `InstanceIfExists` method https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/pkg/cloud/aws/services/ec2/instances.go#L72.
WDYT?

Reply:
why would it be better in the `InstanceIfExists` method? I'm not saying I disagree, but I want to know your thinking behind it

Reply:
Adding the null check in the `InstanceIfExists` method will benefit all callers of that method instead of every caller having to do this check.

Reply:
you can't check the instance id inside of `InstanceIfExists` because there is a panic if the instance id doesn't exist

Reply:
I see. @chuckha What were you thinking?

Reply:
Why not update `InstanceIfExists` to take a pointer? Then you could just pass `scope.MachineStatus.InstanceID` directly and handle the null check?

Reply:
I'll go ahead and update the function signature. I was trying to avoid that but it's probably the cleanest solution

Reply:
@ashish-amarnath I agree with you, but I wanted to hear your logic